### PR TITLE
Add last_used option for pipeline and explicit default

### LINKED ISSUE #17301: Assist action with 'Preferred assistant' opens last used assistant instead
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

After configuring a button with the Assist action and assistant (pipeline) set to 'Preferred assistant', it opens the last used assistant instead of the preferred assistant.

### Describe the behavior you expected

'Preferred assistant' should always use my preferred assistant, even if it's different from the last used assistant.

### Steps to reproduce the issue

1. Have two assistants/pipelines, example: A (preferred) and B.
2. Use assistant B, close the Assist dialog.
3. Create a button card with Assist action, assistant (pipeline) set to 'Preferred assistant'.
4. Click on the button.

Result: assistant B is used.

Expected result: assistant A is used.

### What version of Home Assistant Core has the issue?

core-2023.7.1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Google Chrome 114.0.5735.199 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows 11, version 22H2

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

```yaml
Button card configuration:


show_name: true
show_icon: true
type: button
tap_action:
  action: assist
name: Assist me
```

### Javascript errors shown in your browser console/inspector

```txt
None
```

### Additional information

_No response_

### LINKED ISSUE #17322: Assist (external bus) payload missing details / using implied defaults
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The (external bus) payload that is used for the Assist button in the top app bar and the 'Assist' tap action on a card does not always contain enough details for the app to respond correctly and uses implied defaults which may not match with the app.

Specifically I'm seeing:
- It is implied that if `start_listening` isn't set, Assist shouldn't start listening, while for the app experience it should.
This also translates to the tap action, where if "start listening?" is off it isn't sent on the external bus. There is no way for the app to distinguish this from the Assist action in the top app bar.
- If no specific `pipeline_id` is supplied, it is implied that the last used pipeline _or_ the preferred pipeline is used, depending on where it is triggered - top app bar: last used, action: preferred.
This also applies to the frontend, see #17301. The lack of details here means the frontend always uses the last used pipeline while in the app it always uses the preferred pipeline.

### Describe the behavior you expected

More details, see above

### Steps to reproduce the issue

n/a

### What version of Home Assistant Core has the issue?

core-2023.7.1

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Google Chrome 114.0.5735.199 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows 11, version 22H2

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

```yaml
Payload for top app bar and a button with Assist action as configured below is empty ({}), while they should produce different behavior between frontend/app and top app bar/button.

Button card configuration:
show_name: true
show_icon: true
type: button
tap_action:
  action: assist
name: Assist me
```

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by piitaya**
## Breaking changes

Assist action will open last used pipeline by default instead of preferred to be consistent with toolbar assist.

## Proposed change

Adds new `last_used` option to assist action.

CleanShot 2023-07-17 at 17 14 04

**Trigger assist from the toolbar :** 
It will start listening on app but not on web because web users often have a keyboard on web and it will not work on insecure contexts.
It will use the last used pipeline.

**Trigger assist from the assist action :** 
If the `pipeline_id` is not set, the assist action will use the last used pipeline.
If the `start_listening` option is not set, this assist action will not start listening on web and app.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #17322, fixes #17301
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by balloob — APPROVED** [review]
Approved. Let's wait for @jpelgrom to sign off on these changes before merging.

**Review by jpelgrom — APPROVED** [review]
Code looks good to me, thanks.

**Pull request merged**
